org.openjdk.jmc.console.ext.agent/src/main/java/org/openjdk/jmc/console/ext/agent/manager/wizards/EventMethodParameterEditingPage.java

```diff
@@ -66,13 +95,159 @@ public void createControl(Composite parent) {
+
+		populateStoredValues();
 
 		sc.setExpandHorizontal(true);
 		sc.setExpandVertical(true);
 		sc.setMinSize(container.computeSize(SWT.DEFAULT, SWT.DEFAULT));
 		setControl(sc);
 	}
+
+	private void createPageContent(Composite parent) {
+		GridData gdLabel = new GridData(SWT.BEGINNING, SWT.CENTER, false, false);
+		GridData gdText = new GridData(SWT.FILL, SWT.CENTER, true, true);
+		gdText.minimumWidth = 0;
+		gdText.widthHint = 300;
+
+		Label label = createLabel(parent, "Name:");
+		label.setLayoutData(gdLabel);
+
+		name = createText(parent);
+		name.setMessage("Parameter or Return Value Name");
+		name.setLayoutData(gdText);
+
+		label = createLabel(parent, "Index:");
+		label.setLayoutData(gdLabel);
+
+		Composite indexContainer = new Composite(parent, SWT.NONE);
+		indexContainer.setLayout(new GridLayout(3, false));
+		indexContainer.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+
+		index = new Spinner(indexContainer, SWT.NONE);
+		index.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+
+		label = createLabel(indexContainer, "This is a return value");
+		label.setLayoutData(new GridData(SWT.CENTER, SWT.CENTER, true, true));
+
+		returnValueBtn = new Button(indexContainer, SWT.CHECK);
+		returnValueBtn.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, false, false));
+		returnValueBtn.addListener(SWT.Selection, new Listener() {
+
+			@Override
+			public void handleEvent(Event event) {
+				if (returnValueBtn.getSelection()) {
+					//index.deselectAll();
+					index.setEnabled(false);
+				} else {
+					index.setEnabled(true);
+				}
+
+			}
+
+		});
+
+		label = createLabel(parent, "Description:");
+		label.setLayoutData(gdLabel);
+
+		GridData largeText = new GridData(SWT.FILL, SWT.CENTER, true, true);
+		largeText.minimumWidth = 0;
+		largeText.widthHint = 300;
+		largeText.heightHint = 100;
+		largeText.horizontalSpan = 2;
+
+		description = createTextMulti(parent);
+		description.setMessage("Description of this parameter or return value");
+		description.setLayoutData(largeText);
+
+		label = createLabel(parent, "Content Type:");
+		label.setLayoutData(gdLabel);
+
+		contentType = new Combo(parent, SWT.DROP_DOWN | SWT.READ_ONLY);
+		List<String> contentTypes = new ArrayList<>();
+		for (ICapturedValue.ContentType type : IMethodParameter.ContentType.values()) {
+			contentTypes.add(type.name());
+		}
+		contentType.setItems(contentTypes.toArray(new String[0]));
+		contentType.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+
+		label = createLabel(parent, "Relational Key:");
+		label.setLayoutData(gdLabel);
+
+		relationalKey = createText(parent);
+		relationalKey.setMessage("schema://some-uri");
+		relationalKey.setLayoutData(gdText);
+
+		label = createLabel(parent, "Converter Type:");
+		label.setLayoutData(gdLabel);
+
+		converterType = createText(parent);
+		converterType.setMessage("com.company.project.MyConverter");
```

Comment:
String constants as static class variables, not localizable. Use `org.openjdk.jmc.agent.converters.DefaultStringConverter` instead?

Reply:
On second thought: no, the default should just be empty so that no converter is used.